### ISSUE TITLE
Steam fails to start: fatal stalled cross-thread pipe

### ISSUE THREAD

**Issue by Lucki** [ValveSoftware/steam-for-linux#6752]
#### Your system information

* Steam client version (build number or date): updater built Dec  6 2019 03:34:07
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: Tried both
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Since yesterday (7.12) steam doesn't start anymore. After starting the updater window pops up briefly and closes afterwards. After some minutes without something happening (showing L76 in the log) the process closes itself. I did an update the evening before with some packages involved, downgrading all of them did not get steam back running. There was also an update of steam the day before but I already used that client at that day without problems.
This looks somewhat similar to #5899 but the mentioned fix did not resolve my issue.
I've also tried `steam --reset` which nuked steams folder and did a reinstall but that did not resolve the issue.

Log with gdb backtrace: https://gist.github.com/Lucki/ce42424617a61f5956cc23b20436ffe3

Two lines seems related:
* `../common/pipes.cpp (770) : Assertion Failed: fatal stalled cross-thread pipe`
* `#0  0xf225ea52 in AssertMsgImplementation () from /home/lucki/.local/share/Steam/ubuntu12_32/libtier0_s.so`

Without attaching gdb its uploading a crash dump:
~~~
assert_20191208164523_12.dmp[4952]: Uploading dump (out-of-process)
/tmp/dumps/assert_20191208164523_12.dmp
_ExitOnFatalAssert
assert_20191208164523_12.dmp[4952]: Finished uploading minidump (out-of-process): success = yes
assert_20191208164523_12.dmp[4952]: response: CrashID=bp-7a278ba7-f1ac-46b6-a31e-691d62191208
assert_20191208164523_12.dmp[4952]: file ''/tmp/dumps/assert_20191208164523_12.dmp'', upload yes: ''CrashID=bp-7a278ba7-f1ac-46b6-a31e-691d62191208''
~~~

<details><summary>Updated packages</summary>
Downgrading all of them did not get steam back running:

~~~
[2019-12-07T04:03:35+0100] [ALPM] transaction started
[2019-12-07T04:03:35+0100] [ALPM] upgraded ca-certificates-mozilla (3.47.1-5 -> 3.48-1)
[2019-12-07T04:03:35+0100] [ALPM] upgraded djvulibre (3.5.27-4 -> 3.5.27-5)
[2019-12-07T04:03:35+0100] [ALPM] upgraded gstreamer (1.16.1-1 -> 1.16.2-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded gst-plugins-base-libs (1.16.1-1 -> 1.16.2-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded vulkan-icd-loader (1.1.127-1 -> 1.1.129-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded gst-libav (1.16.1-1 -> 1.16.2-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded openal (1.19.1-2 -> 1.20.0-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded gst-plugins-good (1.16.1-1 -> 1.16.2-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded gst-plugins-bad (1.16.1-4 -> 1.16.2-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded gst-plugins-base (1.16.1-1 -> 1.16.2-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded gst-plugins-ugly (1.16.1-2 -> 1.16.2-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded lib32-gnutls (3.6.10-1 -> 3.6.11.1-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded nspr (4.23-2 -> 4.24-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded lib32-nspr (4.23-1 -> 4.24-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded nss (3.47.1-5 -> 3.48-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded lib32-nss (3.47.1-4 -> 3.48-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded lib32-openal (1.19.1-1 -> 1.20.0-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded lib32-vulkan-icd-loader (1.1.127-1 -> 1.1.129-1)
[2019-12-07T04:03:36+0100] [ALPM] upgraded libmagick6 (6.9.10.74-1 -> 6.9.10.77-1)
[2019-12-07T04:03:42+0100] [ALPM] upgraded linux (5.4.1.arch1-1 -> 5.4.2.arch1-1)
[2019-12-07T04:03:47+0100] [ALPM] upgraded linux-firmware (20191022.2b016af-3 -> 20191118.e8a0f4c-2)
[2019-12-07T04:03:51+0100] [ALPM] upgraded linux-headers (5.4.1.arch1-1 -> 5.4.2.arch1-1)
[2019-12-07T04:03:52+0100] [ALPM] upgraded nvidia-dkms (440.36-4 -> 440.36-5)
[2019-12-07T04:03:52+0100] [ALPM] upgraded opera-ffmpeg-codecs (78.0.3895.5-1 -> 78.0.3904.108-1)
[2019-12-07T04:03:52+0100] [ALPM] upgraded python-peewee (3.12.0-1 -> 3.13.0-1)
[2019-12-07T04:03:52+0100] [ALPM] upgraded seahorse (1:3.34-1 -> 1:3.34-2)
[2019-12-07T04:03:54+0100] [ALPM] upgraded thunderbird (68.2.2-2 -> 68.3.0-1)
[2019-12-07T04:03:54+0100] [ALPM] upgraded thunderbird-i18n-de (68.2.2-1 -> 68.3.0-1)
[2019-12-07T04:03:54+0100] [ALPM] upgraded vhba-module (20190831-23 -> 20190831-24)
[2019-12-07T04:03:55+0100] [ALPM] transaction completed
~~~
</details>

#### Steps for reproducing this issue:

1. Try to run `steam`.
2. Observe a successfully closing process after a few minutes.

**Comment by kisak-valve** [comment]
Hello @Lucki, SIGTRAP is potentially non-fatal, if you tell GDB to continue through it, does it hit a more interesting backtrace soon after that?

Since there's a kernel and nvidia module update in the list of system changes, did you try restarting after the update? It may be worthwhile to check if any other OpenGL application starts fine.

**Comment by Lucki** [comment]
Thanks for you reply. I did a reboot after updating and tried right now Teeworlds and glxgears which both run fine.

I'm not that familiar with gdb so thanks for telling me to continue. Here's the log with continuing following the first log:
<details><summary>Log</summary>

~~~
[…]
#18 0x5657d2c9 in _start ()
(gdb) continue
Continuing.
../common/pipes.cpp (770) : Fatal assert failed: ../common/pipes.cpp, line 770.  Application exiting.

../common/pipes.cpp (770) : Fatal assert failed: ../common/pipes.cpp, line 770.  Application exiting.

_ExitOnFatalAssert
[Thread 0xec1ffb40 (LWP 76483) exited]
[Thread 0xed1cdb40 (LWP 76479) exited]
[Thread 0xef38fb40 (LWP 76471) exited]
[Thread 0xf4dffb40 (LWP 76433) exited]
[Thread 0xf5004b40 (LWP 76430) exited]
[Thread 0xf7962940 (LWP 76426) exited]
[Inferior 1 (process 76426) exited with code 01]
(gdb) bt
No stack.
(gdb) continue
The program is not being run.
~~~
</details>

**Comment by Lucki** [comment]
I was able to figure it out.
Steam tried to access my other harddrive which I used solely as steam library so I haven't noticed that it had somehow a broken partition table. After recovering the [GPT header](https://wiki.archlinux.org/index.php/GPT_fdisk#Recover_GPT_header) I was able to start steam again.
So the problem was on my side, sorry for bothering you. But I wish there was a more useful error message…
